### PR TITLE
Add mixins folder

### PR DESCRIPTION
**Pull request by bquorning**
Closes #942 

I rebased @mockdeep’s pull request #942 but found that I cannot push to that branch. So, I opened this new pull request.

Original description:

This moves several files to a new `mixins/` folder. I was going to add
some new macros, per [this comment][1], but thought it might clutter
things up a bit to put them in the root folder. I thought of adding a
`macros/` folder, but decided to emulate the [rubocop source][2]
instead.

[1]: https://github.com/rubocop-hq/rubocop-rspec/pull/934#issuecomment-643813427
[2]: https://github.com/rubocop-hq/rubocop/tree/master/lib/rubocop/cop/mixin

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [-] Added tests.
* [-] Updated documentation.
* [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: lib/rubocop/cop/rspec/mixin/top_level_describe.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Helper methods for top level describe cops
+      module TopLevelDescribe
```
Comment (pirj) [inline lib/rubocop/cop/rspec/mixin/top_level_describe.rb:7]: I believe we've got rid of this one.

**Review by pirj — APPROVED** [review]
All looks good, except for the retired `top_level_describe`.

File: lib/rubocop/cop/rspec/mixin/top_level_describe.rb (hunk 1 above)
Comment (bquorning) [reply lib/rubocop/cop/rspec/mixin/top_level_describe.rb:7]: You’re absolutely right. I remember thinking about it during the rebase, but I must have forgot. Let’s hope it’s the only thing I forgot 😱

**Comment by bquorning** [comment]
@pirj Please merge at will. #956 will need to be rebased afterwards.

**Pull request merged**